macro-fullcalendar-webjar/pom.xml

```diff
@@ -0,0 +1,116 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+ *
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ *
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.xwiki.contrib</groupId>
+    <artifactId>macro-fullcalendar-pom</artifactId>
+    <version>2.2-SNAPSHOT</version>
+  </parent>
+  <artifactId>macro-fullcalendar-webjar</artifactId>
+  <name>Full Calendar Macro - WebJar</name>
+  <packaging>jar</packaging>
+  <description>Client side resources required by the fullcalendar.io integration in XWiki.</description>
+  <properties>
+    <momentjs.version>2.9.0</momentjs.version>
+    <fullcalendar.version>3.0.0</fullcalendar.version>
+    <webjar.contentDirectory>${project.build.outputDirectory}/META-INF/resources/webjars/${project.artifactId}/${project.version}</webjar.contentDirectory>
+  </properties>
+  <dependencies>
+    <dependency>
+      <groupId>org.webjars</groupId>
+      <artifactId>momentjs</artifactId>
+      <version>${momentjs.version}</version>
+      <scope>runtime</scope>
+    </dependency>
+  </dependencies>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>com.googlecode.maven-download-plugin</groupId>
+        <artifactId>download-maven-plugin</artifactId>
+        <version>1.3.0</version>
+        <executions>
+          <!-- Download the fullcalendar sources (that also contains gcal.js and jquery-ui) for the version tag we
+          want to use. -->
+          <execution>
+            <id>download-fullcalendar</id>
+            <goals>
+              <goal>wget</goal>
+            </goals>
+            <configuration>
+              <url>https://github.com/fullcalendar/fullcalendar/releases/download/v${fullcalendar.version}/fullcalendar-${fullcalendar.version}.zip</url>
```

Comment:
I don't understand why you are packing it yourself. It's available on Maven central according to https://search.maven.org/search?q=g:org.webjars%20AND%20a:fullcalendar&core=gav.

Reply:
I did it because the zip archive contains also the jquery-ui library and I can also minify gcal.js (which is not .min in the jar).
Would it be better to take fullcalendar and jquery-ui from Maven central?

Reply:
Yes, I think it would be better to use the available jars from Maven central. But note that jquery-ui is also a core extension since 6.3M2 (I think, see [XWIKI-10333](https://jira.xwiki.org/browse/XWIKI-10333)) so make sure you don't specify a version greater than 1.11.1 (for 6.3M2, or 1.11.4 for 7.3RC1).